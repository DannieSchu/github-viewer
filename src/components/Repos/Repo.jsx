import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ name, html_url, language, created_at, updated_at }) => {
  return (
    <>
      <a href={html_url} target="blank">{name}</a>
      <p>Language: {language}</p>
      <p>Created: {created_at}</p>
      <p>Updated: {updated_at}</p>
    </>
  );
};

Repo.propTypes = {
  name: PropTypes.string.isRequired,
  html_url: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  created_at: PropTypes.string.isRequired,
  updated_at: PropTypes.string.isRequired
};

export default Repo;
