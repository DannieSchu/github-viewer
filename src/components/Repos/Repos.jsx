import React from 'react';
import PropTypes from 'prop-types';
import Repo from './Repo.jsx';

const Repos = ({ repos }) => {
  const individualRepos = repos.map(repo => (
    <li key={repo.id}>
      <Repo {...repo} />
    </li>
  ));

  return (
    <ul>
      {individualRepos}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    repoName: PropTypes.string.isRequired,
    repoUrl: PropTypes.string.isRequired,
    language: PropTypes.string.isRequired,
    dateCreated: PropTypes.string.isRequired,
    dateUpdated: PropTypes.string.isRequired
  })).isRequired
};

export default Repos;
