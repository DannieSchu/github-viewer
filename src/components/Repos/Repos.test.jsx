import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos.jsx';

describe('Repos', () => {
  it('matches a snapshot', () => {
    const repos = [{      
      repoName: 'About-Me',
      id: 3529521,
      repoUrl: 'https://github.com/DannieSchu/About-Me',
      language: 'JavaScript',
      dateCreated: '2020-01-14T00:49:15Z',
      dateUpdated: '2020-01-14T00:49:15Z'
    }, {
      repoName: 'Bus-Mall',
      id: 846732,
      repoUrl: 'https://github.com/DannieSchu/Bus-Mall',
      language: 'JavaScript',
      dateCreated: '2020-01-14T00:49:15Z',
      dateUpdated: '2020-01-14T00:49:15Z'
    }];

    const wrapper = shallow(
      <Repos repos={repos} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
