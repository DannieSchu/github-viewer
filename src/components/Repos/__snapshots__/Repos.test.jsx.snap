// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos matches a snapshot 1`] = `
<ul>
  <li
    key="3529521"
  >
    <Repo
      dateCreated="2020-01-14T00:49:15Z"
      dateUpdated="2020-01-14T00:49:15Z"
      id={3529521}
      language="JavaScript"
      repoName="About-Me"
      repoUrl="https://github.com/DannieSchu/About-Me"
    />
  </li>
  <li
    key="846732"
  >
    <Repo
      dateCreated="2020-01-14T00:49:15Z"
      dateUpdated="2020-01-14T00:49:15Z"
      id={846732}
      language="JavaScript"
      repoName="Bus-Mall"
      repoUrl="https://github.com/DannieSchu/Bus-Mall"
    />
  </li>
</ul>
`;
